src/utils/starknet_compilation.py

```diff
@@ -72,3 +73,17 @@ def compile_contract(
         )
         assert isinstance(assembled, ContractDefinition)
         return assembled
+
+    class AbiElement(TypedDict):
+        """NOTE: This type doesn't represent a dictionary faithfully. The dictionary can have more attributes."""
+
+        name: str
+        type: Literal["function"]
```

Comment:
Why do we need this class?, can't we just use string and make aliast `FunctionName = str`?

Reply:
type is always function, so we basically introducing misleading abstraction which does kinda nothing

Reply:
I'll change the type's type to string then. The goal of using Literal was expanding this type when necessary.

`StarknetCompiler::get_functions` uses "type" attribute of this dictionary. It looks like only function name is used in the remaining part of the code.

I'll bring back the dict type in this PR. In the other PR I'll change `test_functions` to `test_functino_names`.